components/webext-storage/src/api.rs

```diff
@@ -332,6 +332,45 @@ pub fn get_bytes_in_use(conn: &Connection, ext_id: &str, keys: JsonValue) -> Res
     Ok(size)
 }
 
+/// Information about the usage of a single extension.
+#[derive(Debug, Clone, PartialEq)]
+pub struct UsageInfo {
+    /// The extension id.
+    pub extension: String,
+    /// The number of keys the extension uses.
+    pub num_keys: usize,
+    /// The number of bytes used by the extension. This result is somewhat rough
+    /// -- it doesn't bother counting the size of the extension ID, or data in
+    /// the mirror, and favors returning the exact number of bytes used by the
+    /// column (that is, the size of the JSON object) rather than replicating
+    /// the `get_bytes_in_use` return value for all keys.
```

Comment:
I had to double-check how `get_bytes_in_use` calculates usage...this makes sense to me as the size of the JSON string in the `data` column, but the comment did throw me for a loop at first.

Reply:
Right, it's a bit wonky. I think we care more about this value than the other (and it's also easier to compute >_>)